From 47cc9432a63d75d5a82394b75f6c4529277b14c4 Mon Sep 17 00:00:00 2001
From: Ben Noordhuis <info@bnoordhuis.nl>
Date: Tue, 31 Dec 2019 14:55:32 +0100
Subject: [PATCH] Fix inspector protocol generation with CMake

Preserve absolute inspector protocol paths passed by CMake. GN apparently
passes relative paths, but relying on that is not robust.

Import Mapping from collections.abc so V8's vendored Jinja2 works with
Python 3.10 and newer.
---
 v8/third_party/inspector_protocol/code_generator.py | 2 +-
 v8/third_party/jinja2/tests.py                       | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/v8/third_party/inspector_protocol/code_generator.py b/v8/third_party/inspector_protocol/code_generator.py
index 4260dbe..50e439e 100755
--- a/v8/third_party/inspector_protocol/code_generator.py
+++ b/v8/third_party/inspector_protocol/code_generator.py
@@ -80,7 +80,7 @@ def read_config():
     config_file = arg_options.config
     config_base = os.path.dirname(config_file)
     config_values = arg_options.config_value
-    inspector_protocol_dir = arg_options.inspector_protocol_dir.lstrip('/')
+    inspector_protocol_dir = arg_options.inspector_protocol_dir
   except Exception:
     # Work with python 2 and 3 http://docs.python.org/py3k/howto/pyporting.html
     exc = sys.exc_info()[1]
diff --git a/v8/third_party/jinja2/tests.py b/v8/third_party/jinja2/tests.py
--- a/v8/third_party/jinja2/tests.py
+++ b/v8/third_party/jinja2/tests.py
@@ -10,7 +10,7 @@
 """
 import operator
 import re
-from collections import Mapping
+from collections.abc import Mapping
 from jinja2.runtime import Undefined
 from jinja2._compat import text_type, string_types, integer_types
 import decimal
-- 
2.20.1

